File: Python version/specializations/clingo/desktop/clingo_desktop_service.py
from platforms.desktop.desktop_service import DesktopService
from specializations.clingo.clingo_answer_sets import ClingoAnswerSets


class ClingoDesktopService(DesktopService):
    """Extention of DesktopService for Clingo."""

    def __init__(self, exe_path):
        super(ClingoDesktopService, self).__init__(exe_path)
        self._load_from_stdin_option = "-"

    def _get_output(self, output, error):
        """Returns a new ClingoAnwerSets from given output and error."""
        return ClingoAnswerSets(output, error)

    def start_sync(self, programs, options):
        """Calls start_sync method of a superclass and returns its output
        object."""
        return super(ClingoDesktopService, self).start_sync(programs, options)

    def start_async(self, callback, programs, options):
        """Calls start_async method of a superclass."""
        super(ClingoDesktopService, self).start_async(
            callback, programs, options)

File: Java version/app/src/main/java/it/unical/mat/embasp/specializations/clingo/desktop/ClingoDesktopService.java
package it.unical.mat.embasp.specializations.clingo.desktop;

import it.unical.mat.embasp.base.Output;
import it.unical.mat.embasp.platforms.desktop.DesktopService;
import it.unical.mat.embasp.specializations.clingo.ClingoAnswerSets;

/** is an extention of {@link DesktopService} for Clingo 's solver */

public class ClingoDesktopService extends DesktopService {

	public ClingoDesktopService(final String exe_path) {
		super(exe_path);
		load_from_STDIN_option = "";
	}

	@Override
	protected Output getOutput(final String output, final String error) {
		return new ClingoAnswerSets(output, error);
	}

}
